cmd/generic-handlers.go

```diff
@@ -469,3 +477,77 @@ func (h pathValidityHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	h.handler.ServeHTTP(w, r)
 }
+
+// logginHandler logs all HTTP requests and responses
+// and write the log to a io.Writer object.
+type loggingHandler struct {
+	writer    io.Writer
+	writeLock sync.Mutex
+	handler   http.Handler
+}
+
+func (h loggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Time that will be printed in log
+	timeFormat := "2006-01-02 15:04:05 -0700"
+	// Generate short random request ID
+	randBytes := []byte("0000")
+	_, _ = io.ReadFull(rand.Reader, randBytes[:])
+	reqID := fmt.Sprintf("%X", randBytes)
```

Comment:
IMO we could use `time.Now().UnixNano()` as request ID.  This is better than random number and better debugability.